semanticdb/metacp/src/main/scala/scala/meta/internal/scalacp/Scalalib.scala

```diff
@@ -48,11 +48,34 @@ object Scalalib {
     builtinClass(List(p.ABSTRACT, p.FINAL), "Null", List("scala.AnyRef#"), Nil)
   }
 
+  def singletonTrait: ToplevelInfos = {
+    builtinTrait(List(p.FINAL), "Singleton", List("scala.Any#"), Nil)
+  }
+
   private def builtinClass(
       props: List[s.SymbolInformation.Property],
       name: String,
       bases: List[String],
-      symbols: List[s.SymbolInformation]): ToplevelInfos = {
+      symbols: List[s.SymbolInformation]
+  ): ToplevelInfos = {
+    builtinClassOrTrait(props, name, bases, symbols, isTrait = false)
+  }
+
+  private def builtinTrait(
+      props: List[s.SymbolInformation.Property],
+      name: String,
+      bases: List[String],
+      symbols: List[s.SymbolInformation]
+  ): ToplevelInfos = {
+    builtinClassOrTrait(props, name, bases, symbols, isTrait = true)
+  }
+
+  private def builtinClassOrTrait(
+      props: List[s.SymbolInformation.Property],
+      name: String,
+      bases: List[String],
+      symbols: List[s.SymbolInformation],
+      isTrait: Boolean): ToplevelInfos = {
```

Comment:
can we replace boolean with `Kind`? That should allow us to have a single `builtin` method

Reply:
👍 sounds right. This is where I really miss union types:

```scala
private def builtinClassOrTrait(
    kind: k.CLASS | k.TRAIT,
    props: List[s.SymbolInformation.Property],
    name: String,
    bases: List[String],
    symbols: List[s.SymbolInformation]): ToplevelInfos